src/models/storage.ts

```diff
@@ -39,14 +39,26 @@ class EntryStorage {
     return newData;
   }
 
+  private static ensureObject(
+      _data: {[hash: string]: OTPStorage}, hash: string) {
+    if (typeof _data[hash] !== 'object') {
```

Comment:
`_data[hash]` may not exist (we cannot ensure `hash` is gotten from `hash of Object.keys(_data)`, though we do it currently), `hasOwnProperty` should be a good check here.

Reply:
Also, we may check more here. `_data[hash]` is not only an `object`, but also should be `OTPEntry`. So we can check if it has the required fields for `OTPEntry`, such as `type`, `secret`, etc.